src/azure-cli/azure/cli/command_modules/appconfig/_kv_helpers.py

```diff
@@ -281,6 +281,19 @@ def __write_kv_and_features_to_file(file_path, key_values=None, features=None, f
         raise CLIError("Failed to export key-values to file. " + str(exception))
 
 
+# Exported in the format @Microsoft.AppConfiguration(Endpoint=<storeEndpoint>; Key=<kvKey>; Label=<kvLabel>).
+# Label is optional
+
+def __map_to_appservice_config_reference(key_value, endpoint, prefix):
+    label = key_value.label
+    key_value.value = AppServiceConstants.APPSVC_CONFIG_REFERENCE_PREFIX + 'Endpoint={0}; Key={1}'.format(
+        endpoint, key_value.key) + ('; Label={0}'.format(label) if label is not None else '') + ')'
+
+    if key_value.key.startswith(prefix):
+        key_value.key = key_value.key[len(prefix):]
+    return key_value
```

Comment:
In `__write_kv_to_app_service`, we check if the content type is that of KeyVault reference and try to convert the value to AppService format KeyVault reference. But if we have already converted a KeyVault reference to an AppConfig reference here, the check in  `__write_kv_to_app_service` is unnecessarily going to raise an exception while deserializing the value. Although the exception is not thrown and the value is treated as regular string value, we should avoid doing things that will definitely raise exceptions, because catching an exception is expensive.

Can you check if there are any side effects if we clear the content type for all key-values here? I dont think we need the content type while exporting to AppService after this point.

Reply:
I confirmed that the content type is not used when exporting to app service. So this can be changed without any side effects. Technically, the key-value is no more a key-vault reference.  

Another option would be to supply the `export-as-reference` flag to the `__write_kv_to_app_service` method and skip the key-vault reference validation when it is true.

Reply:
I'd prefer changing the content type, because as you said - its technically not a KeyVault reference anymore.